server/src/main/webapp/WEB-INF/rails/webpack/views/pages/elastic_agents/wizard.tsx

```diff
@@ -0,0 +1,618 @@
+      result         = this.findPluginWithId(vnode, pluginId);
+    }
+
+    if (!result) {
+      result = vnode.attrs.pluginInfos()[0];
+    }
+
+    return result;
+  }
+
+  private findPluginWithId(vnode: m.Vnode<Attrs>, pluginId?: string) {
+    return vnode.attrs.pluginInfos().find((pluginInfo) => pluginInfo.id === pluginId);
+  }
+
+  private clusterProfileForm(vnode: m.Vnode<Attrs, this>) {
+    const elasticAgentSettings = this.pluginInfoForCurrentClusterProfileOrDefaultToFirstPlugin(vnode)
+                                     .extensionOfType<ElasticAgentExtension>(ExtensionTypeString.ELASTIC_AGENTS)!;
+    return (
+      <AngularPluginNew pluginInfoSettings={Stream(elasticAgentSettings.clusterProfileSettings)}
+                        configuration={vnode.attrs.clusterProfile().properties()}
+                        key={vnode.attrs.clusterProfile().pluginId()}
+                        disabled={vnode.attrs.readonly}/>
+    );
+  }
+}
+
+class ElasticProfileWidget extends MithrilViewComponent<Attrs> {
+  view(vnode: m.Vnode<Attrs, this>) {
+    const elasticAgentExtension        = this.pluginInfoForCurrentElasticProfileOrDefaultToFirstPlugin(vnode)
+                                             .extensionOfType<ElasticAgentExtension>(ExtensionTypeString.ELASTIC_AGENTS)!;
+    const elasticProfileConfigurations = elasticAgentExtension.profileSettings;
+
+    return (
+      <div class={styles.container}>
+        <div>
+          <div class={styles.profileForm}>
+            <TextField label="Elastic Profile Name"
+                       property={vnode.attrs.elasticProfile().id}
+                       errorText={vnode.attrs.elasticProfile().errors().errorsForDisplay("id")}
+                       readonly={vnode.attrs.etag !== undefined}
+                       required={true}/>
+          </div>
+          <div>
+            <div
+              className={foundationClassNames(foundationStyles.foundationGridHax, foundationStyles.foundationFormHax)}>
+              <div class="row collapse" data-test-id="properties-form">
+                <AngularPluginNew
+                  pluginInfoSettings={Stream(elasticProfileConfigurations)}
+                  configuration={vnode.attrs.elasticProfile().properties()}
+                  key={vnode.attrs.elasticProfile().pluginId}/>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div>
+          <ConceptDiagram image={elasticAgentProfileImg} css={conceptDiagramStyles}>
+            <h2>What is an Elastic Agent Profile?</h2>
+            <p>An Elastic Agent Profile usually contains the configuration for your agent.<br/>
+              Depending on the plugin used, this may contain the machine image (ami, docker image), size of the
+              CPU/memory/disk, network settings among other things.</p>
+          </ConceptDiagram>
+        </div>
+      </div>
+    );
+  }
+
+  private findPluginWithId(vnode: m.Vnode<Attrs>, pluginId?: string) {
```

Comment:
Suggestion:

We can use `findByPluginId` method on `PluginInfos` instead of this.